app/__init__.py

```diff
@@ -66,33 +68,40 @@
 prefixed_store = PrefixDecorator('sessions_', store)
 KVSessionExtension(prefixed_store, app)
 
-url_prefix = app.config['SERVICE_PREFIX']
-blueprints = (
-    gitlab_auth.blueprint,
-    jupyterhub_auth.blueprint,
-    web.blueprint,
-    graph.blueprint,
-    notebooks.blueprint,
-    jupyterhub.blueprint,
-    webhooks.blueprint,
-    gitlab.blueprint,
-)
-
 
 @app.route('/health', methods=['GET'])
 async def healthcheck():
     return Response(json.dumps("Up and running"), status=200)
 
 
-def _join_url_prefix(*parts):
-    """Join prefixes."""
-    parts = [part.strip('/') for part in parts if part and part.strip('/')]
-    if parts:
-        return '/' + '/'.join(parts)
+def add_blueprints(app):
```

Comment:
if you use `current_app` you won't have a problem with circular dependencies and you can keep it on global level.